src/reducetensor.cpp

```diff
@@ -267,6 +285,124 @@ inline int GetReduceTensorOpId(miopenReduceTensorOp_t t)
     };
 };
 
+static std::string get_network_config_string_from_type_enums(miopenDataType_t TSrc,
+                                                             miopenDataType_t TComp,
+                                                             miopenDataType_t TDst)
+{
+    std::ostringstream outs;
+
+    outs << TSrc << TComp << TDst;
+
+    return (outs.str());
+};
+
+static std::string get_definition_string_from_type_enums(miopenDataType_t TSrc,
```

Comment:
I will use the same converter used by static reduction to convert from miopen::miopenDataType_t to the data types used by the dynamic reduction kernel.   If doing so,  Dynamic reduction kernel does not need ck::DataTypeEnum_t.

Reply:
Please write a converter between ```miopen::miopenDataType_t`` into ```ck::DataTypeEnum_t``` for dynamic kernel.

Reply:
Please keep these in mind:
1. make all the host logic designed for dynamic kernel, NOT static kernel
2. If the logic of static kernel and dynamic kernel are different, DO NOT mix the logic in the same function, put them in different functions.
3. If static kernel can reuse the host logic designed for dynamic kernel, it's OK to reuse it. If not, write a separate logic in separate function

The reason is that we want to keep iterating on the implementation of dynamic kernels (both kernels and solvers), and then fully retire static kernel. So we don't want to mix their the logic of them together. Everything about the refactor should be making their logic more separate from each other

Reply:
Should we use ck::DataTypeEnum_t ?    I found no reason to use this,   since we can pass the types using a consistent representation, like `D` for double, `F` for float,   and convert these consistent representation to types directly.   We don't work on Type Enum in the kernel, so no need.